core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala

```diff
@@ -610,6 +610,51 @@ class GroupMetadataManagerTest {
   }
 
   @Test
+  def testLoadGroupAndOffsetsFromDifferentSegments(): Unit = {
```

Comment:
I don't want to debate, I want this to be fixed, because we have this issue on some of our systems. I will change issue title, commit message and PR title tomorow morning, but I think it is too complicated to reproduce the single segment multiple read case.